src/app/sessions/models/session-edit-data.ts

```diff
@@ -0,0 +1,9 @@
+import { SessionCreate } from './session-create.model';
+import { SessionEditOrCreateDialogComponent } from '../components/session-edit-or-create-dialog/session-edit-or-create-dialog.component';
+import { MatDialogRef } from '@angular/material';
+
+export interface SessionEditData {
```

Comment:
SessionEditData contains a callback and dialog reference. Seems like stuff got mixed. I would not include the onCancel and DialogReference in this model which would leave it only with SessionCreate, which then makes SessionEditData not needed